File: fetch_tr.py
import sys
from youtube_transcript_api import YouTubeTranscriptApi

try:
    transcript = YouTubeTranscriptApi.get_transcript('EUstAnAAEfI', languages=['fr', 'ar', 'en'])
    text = " ".join([i['text'] for i in transcript])
    print(text)
except Exception as e:
    print(f"Error fetching transcript: {e}")

File: fetch_tr.js
import { YoutubeTranscript } from 'youtube-transcript';

YoutubeTranscript.fetchTranscript('EUstAnAAEfI', { lang: 'fr' })
  .then(transcript => {
    console.log(transcript.map(t => t.text).join(' '));
  })
  .catch(err => {
    // try fallback without lang
    YoutubeTranscript.fetchTranscript('EUstAnAAEfI')
      .then(transcript => {
        console.log(transcript.map(t => t.text).join(' '));
      })
      .catch(console.error);
  });
